test/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,90 @@
+describe('axe.utils.preloadCssom unit tests', function() {
+	'use strict';
+
+	var args;
+
+	function addStyleToHead() {
+		var css = 'html {font-size: inherit;}';
+		var head = document.head || document.getElementsByTagName('head')[0];
+		var style = document.createElement('style');
+		style.id = 'preloadCssomTestHeadSheet';
+		style.type = 'text/css';
+		style.appendChild(document.createTextNode(css));
+		head.appendChild(style);
+	}
+
+	function removeStyleFromHead() {
+		var s = document.getElementById('preloadCssomTestHeadSheet');
+		if (s) {
+			s.parentNode.removeChild(s);
+		}
+	}
+
+	beforeEach(function() {
+		addStyleToHead();
+		args = {
+			asset: 'cssom',
+			timeout: 10000,
+			treeRoot: (axe._tree = axe.utils.getFlattenedTree(document))
+		};
+	});
+
+	afterEach(function() {
+		removeStyleFromHead();
+	});
+
+	it('should be a function', function() {
+		assert.isFunction(axe.utils.preloadCssom);
+	});
+
+	it('should return a queue', function() {
+		var actual = axe.utils.preloadCssom(args);
+		assert.isObject(actual);
+		assert.containsAllKeys(actual, ['then', 'defer', 'catch']);
+	});
+
+	it('should ensure result of cssom is an array of sheets', function(done) {
+		var actual = axe.utils.preloadCssom(args);
+		actual
+			.then(function(results) {
+				assert.lengthOf(results[0], 2); // returned from queue, hence the index look up
+				done();
+			})
+			.catch(function(error) {
+				done(error);
+			});
+	});
+
+	it('should fail if number of sheets returned does not match stylesheets defined in document', function(done) {
```

Comment:
This isn't what's being tested. It's not true either, this should return styles in shadow DOM too.

Reply:
Done, ignoring shadowDOM in this PR.

Reply:
Sorted!